src/main/core/support/configuration.rs

```diff
@@ -85,26 +85,25 @@ pub struct ConfigOptions {
 }
 
 impl ConfigOptions {
-    pub fn new(config_file: &ConfigFileOptions, options: &CliOptions) -> Self {
-        let mut config_file = config_file.clone();
-
+    pub fn new(mut config_file: ConfigFileOptions, options: CliOptions) -> Self {
         // override config options with command line options
-        config_file.general = GeneralOptions::merge(&options.general, &config_file.general);
+        config_file.general = GeneralOptions::merge(options.general, config_file.general);
         config_file.host_defaults =
-            HostDefaultOptions::merge(&options.host_defaults, &config_file.host_defaults);
+            HostDefaultOptions::merge(options.host_defaults, config_file.host_defaults);
         config_file.experimental =
-            ExperimentalOptions::merge(&options.experimental, &config_file.experimental);
+            ExperimentalOptions::merge(options.experimental, config_file.experimental);
 
         // copy the host defaults to all of the hosts
         for (_, host) in &mut config_file.hosts {
-            host.options = HostDefaultOptions::merge(&host.options, &config_file.host_defaults);
+            host.options =
+                HostDefaultOptions::merge(host.options.clone(), config_file.host_defaults.clone());
```

Comment:
The `host.options.clone()` is needed because `host` is behind a mutable reference. Rust doesn't currently allow you to iterate over values without consuming those values.